pydrive2/auth.py

```diff
@@ -367,25 +357,55 @@ def LoadCredentialsFile(self, credentials_file=None):
         :raises: InvalidConfigError, InvalidCredentialsError
         """
         if credentials_file is None:
-            self._default_storage = self._storages["file"]
-            if self._default_storage is None:
+            credentials_file = self.settings.get("save_credentials_file")
+            if credentials_file is None:
                 raise InvalidConfigError(
-                    "Backend `file` is not configured, specify "
-                    "credentials file to read in the settings "
-                    "file or pass an explicit value"
+                    "Please specify credentials file to read"
                 )
-        else:
-            self._default_storage = Storage(credentials_file)
 
         try:
-            self.credentials = self._default_storage.get()
+            auth_info = self.storage.read_credentials(credentials_file)
+        except FileNotFoundError:
+            # if credential found was not found, raise the error for handling
+            raise
         except OSError:
+            # catch other errors
             raise InvalidCredentialsError(
                 "Credentials file cannot be symbolic link"
             )
 
-        if self.credentials:
-            self.credentials.set_store(self._default_storage)
+        try:
+            self._credentials = google.oauth2.credentials.Credentials.from_authorized_user_info(
+                auth_info, scopes=auth_info["scopes"]
+            )
+        except ValueError:
+            # if saved credentials lack a refresh token
+            # handled for backwards compatibility
+            warn(
+                "Loading authorized user credentials without a refresh token is "
+                "not officially supported by google auth library. We recommend that "
+                "you only store refreshable credentials moving forward."
+            )
+
+            self._credentials = google.oauth2.credentials.Credentials(
+                token=auth_info.get("token"),
+                token_uri="https://oauth2.googleapis.com/token",  # always overrides
+                scopes=auth_info.get("scopes"),
+                client_id=auth_info.get("client_id"),
+                client_secret=auth_info.get("client_secret"),
+            )
+
+        # in-case reauth / consent required
+        # create a flow object so that reauth flow can be triggered
+        additional_config = {}
```

Comment:
Also, don't quite understand this. How was this handled before?

Reply:
I mean the whole `create a flow object so that reauth flow can be triggered` part

Reply:
In the past,  `self.flow` was populated using `self.GetFlow()` as part of the `CheckAuth` decorator, which handles the whole refresh process. Now that refresh is handled by the `google_auth` lib, I've opted to create the Flow object using the recently loaded credentials

Reply:
but you also have that lazy loading property `flow` that is being used internally, or do you mean that we have to create it again in certain cases?

also, btw, I think `flow()` now duplicates `GetFlow` which is not nice.